core/frontend/src/components/vehiclesetup/configuration/compass/ArdupilotMavlinkCompassSetup.vue

```diff
@@ -309,15 +325,64 @@ export default Vue.extend({
           (name) => autopilot_data.parameter(name),
         )
         if (offset_params.includes(undefined) || diagonal_params.includes(undefined)) {
-          results[compass.param] = false
+          results[compass.param] = {
+            calibration: CalibrationType.UNKNOWN,
+            color: 'red',
+            alert: 'error',
+            description: 'Unable to get calibration status, something went wrong.',
+            calibration_short: 'Unknown',
+          }
           continue
         }
-        const scale_param_name = `COMPASS_SCALE${compass_number}`
-        const scale_param = autopilot_data.parameter(scale_param_name)
         const is_at_default_offsets = offset_params.every((param) => param?.value === 0.0)
+        if (is_at_default_offsets) {
+          results[compass.param] = {
+            calibration: CalibrationType.NOT_CALIBRATED,
+            color: 'red',
+            alert: 'error',
+            description: 'The compass is not calibrated, please calibrate it.',
+            calibration_short: 'Calibration Needed',
+          }
+
+          continue
+        }
+
         const is_at_default_diagonals = diagonal_params.every((param) => param?.value === 0.0)
-        results[compass.param] = offset_params.isEmpty() || diagonal_params.isEmpty()
-          || !is_at_default_offsets || !is_at_default_diagonals || scale_param?.value !== 0.0
+        if (is_at_default_diagonals) {
+          results[compass.param] = {
+            calibration: CalibrationType.QUICK,
+            color: 'orange',
+            alert: 'warning',
+            description: 'Quick Calibration is easier for large vehicle, but has the poorer '
+  + 'performance (consider using log-based calibration afterwards).',
+            calibration_short: 'Calibrated (Quick)',
+          }
+          continue
+        }
+
+        const scale_param_name = `COMPASS_SCALE${compass_number}`
+        const scale_param = autopilot_data.parameter(scale_param_name)
+        const is_at_default_scale = scale_param?.value === 0.0
+
+        if (is_at_default_scale) {
+          results[compass.param] = {
+            calibration: CalibrationType.FULL_NO_WMM,
+            color: 'orange',
```

Comment:
Can this use `var(--warning)` from our theme stuff? And similar for the other colours?

Reply:
done =]